extension/observer/k8sobserver/config.go

```diff
@@ -36,6 +38,14 @@ type Config struct {
 	//
 	// Then set this value to ${K8S_NODE_NAME} in the configuration.
 	Node string `mapstructure:"node"`
+	// ObservePods determines whether to report observer pod and port endpoints. If `true` and Node is specified
+	// it will only discover pod and port endpoints whose `spec.nodeName` matches the provided node name. If `true` and
+	// Node isn't specified, it will discover all available pod and port endpoints. `true` by default.
+	ObservePods bool `mapstructure:"observe_pods"`
+	// ObserveNodes determines whether to report observer k8s.node endpoints. If `true` and Node is specified
+	// it will only discover node endpoints whose `metadata.name` matches the provided node name. If `true` and
+	// Node isn't specified, it will discover all available node endpoints. `false` by default.
+	ObserveNodes bool `mapstructure:"observe_nodes"`
```

Comment:
either `observe_pods` or `observe_nodes` has to be true right? Should we add a validation for that?

Reply:
Not necessarily but a silent noop doesn't seem desirable. Will add a validation step.